src/pbt/v2/client/airflow_client/mwaa_client.py

```diff
@@ -0,0 +1,155 @@
+
+    def unpause_dag(self, dag_id: str) -> DAG:
+        self.get_dag(dag_id)
+        response_as_text = self._get_response(f"dags unpause {dag_id}")
+        response_as_json = self._clean_response(response_as_text)
+        print(f"Response as json {response_as_json}")
+        return DAG.create_from_mwaa(response_as_json)
+
+    def upload_dag(self, dag_id: str, file_path: str):
+        relative_path = f"{self.dag_s3_path}/{dag_id}.zip"
+        try:
+            self.s3_aws_client.upload_file(file_path, self._source_bucket, relative_path)
+        except Exception as e:
+            print(f"Error uploading file {file_path} to bucket {self._source_bucket}", e)
+            raise DagUploadFailedException(f"Error uploading file {file_path} to bucket {self._source_bucket}", e)
+
+    @retry((DagNotAvailableException,), total_tries=50, delay_in_seconds=10, backoff=0)
+    def get_dag(self, dag_id: str) -> DAG:
+        response = self._get_response(f"dags list -o json")
+        dag_list = json.loads(response)
+        dag = next((dag for dag in dag_list if dag['dag_id'] == dag_id and dag['paused'] is not None), None)
+
+        if dag is not None:
+            return DAG.create_from_mwaa(dag)
+        else:
+            raise DagNotAvailableException(f"Dag {dag_id} not found")
+
+    def delete_dag(self, dag_id: str) -> str:
+        response = self._get_response(f"dags delete {dag_id} --yes")
+        return response
+
+    def _create_cli_token(self):
+        return self.mwaa_client.create_cli_token(Name=self.environment_name)
+
+    def _expiry_cli_token(self):
+        return ExpiringValue(lambda: self._create_cli_token(), 60 * 60 * 24)
+
+    def _headers(self):
+        return {
+            "Authorization": f"Bearer {self._expiry_cli_token().get_value()['CliToken']}",
+            "User-Agent": "Prophecy"
+        }
+
+    def _clean_response(self, text):
+        try:
+            return json.loads(text)  # Try to parse the text as JSON
+        except json.JSONDecodeError:
+            # If unsuccessful, perform string transformations and return the result
+            cleaned_text = "{\"" + text.strip() \
+                .replace(", ", "\", \"") \
+                .replace(": ", "\":\"") \
+                .replace(",\"", ", \"") + "\"}"
+            return json.loads(cleaned_text)
+
+    def _execute_airflow_command(self, command: str):
+        print(f"Expiry cli token {self._expiry_cli_token().get_value()}")
+        url = f"https://{self._expiry_cli_token().get_value()['WebServerHostname']}/aws_mwaa/cli"
+        response = requests.post(url, headers=self._headers(), data=command)
+        response.raise_for_status()  # Raise an HTTPError if the status is 4xx, 5xx
+        return response.text  # Get the response body as text
+
+    def _get_response(self, command: str):
+        response_body = self._execute_airflow_command(command)
+
+        decoded_response = json.loads(response_body)
```

Comment:
why can't we call response.json in the above method?

Reply:
it's the structure 
it's a json of encoded string json. 

that's why response.json will not work.